lib/rexml/xpath_parser.rb

```diff
@@ -78,7 +78,15 @@ def variables=( vars={} )
 
     def parse path, nodeset
       path_stack = @parser.parse( path )
-      match( path_stack, nodeset )
+      if nodeset.first.node_type == :attribute
```

Comment:
Could you care `nodeset.empty?` case?

Can we remove this check by adding `Attribute#document`?

Reply:
OK
I see.